File: code/349.intersection-of-two-arrays.py
#
# @lc app=leetcode id=349 lang=python3
#
# [349] Intersection of Two Arrays
#

# @lc code=start
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        ret = list[int]()
        set1 = set(nums1)
        set2 = set(nums2)
        for num in set1:
            if num in set2:
                ret.append(num)

        return ret

# @lc code=end

File: code/done/349.intersection-of-two-arrays.go
/*
 * @lc app=leetcode id=349 lang=golang
 *
 * [349] Intersection of Two Arrays
 */

// @lc code=start
package main

func intersection(nums1 []int, nums2 []int) []int {
	result := []int{}
	set := make(map[int]int)
	for _, v := range nums1 {
		set[v] = 1
	}
	for _, v := range nums2 {
		if _, ok := set[v]; ok {
			result = append(result, v)
			delete(set, v)
		}
	}
	return result
}

// @lc code=end
